### PR TITLE
[MM-34305] Migrate 'components/admin_console/admin_user_card' and associated tests to TypeScript

### PR DESCRIPTION
**Pull request by arvinDarmawan**
Summary
This code migrate "components/admin_console/admin_user_card" component to TypeScript.

Ticket Link
Jira card: https://mattermost.atlassian.net/browse/MM-34305
Github issue: https://github.com/mattermost/mattermost-server/issues/17245

```release-note
Release Note
- Refactor component to typescript, no changes in UI
```

### CODE REVIEW

File: components/admin_console/admin_user_card/admin_user_card.tsx [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+
+import {Client4} from 'mattermost-redux/client';
+
+import ProfilePicture from 'components/profile_picture';
+import * as Utils from 'utils/utils.jsx';
+
+import './admin_user_card.scss';
+
+type BulletProps = {
+    user: {
```
Comment (hmhealey) [inline components/admin_console/admin_user_card/admin_user_card.tsx:14]: Can these user props use the `UserProfile` type from `mattermost-redux/types/users`? If you get complains from the unit tests not passing in properly typed objects, you can use the `TestHelper.getUserMock` method to construct proper user objects.

File: components/admin_console/admin_user_card/admin_user_card.tsx [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+
+type Props = {
+    user: {
+        first_name: string | null;
+        last_name: string | null;
+        nickname: string | null;
+        last_picture_update?: number | undefined;
+        id: string;
+    };
+    body?: JSX.Element | undefined;
```
Comment (hmhealey) [inline components/admin_console/admin_user_card/admin_user_card.tsx:30]: `body` and `footer` can actually be typed `React.ReactNode`. That type just includes everything renderable in React, so it has `JSX.Element`, `undefined`, strings, and so on are all included.

File: packages/mattermost-redux/src/client/client4.ts [hunk 3]
```diff
@@ -870,7 +870,7 @@ export default class Client4 {
         );
     };
 
-    getProfilePictureUrl = (userId: string, lastPictureUpdate: number) => {
+    getProfilePictureUrl = (userId: string, lastPictureUpdate?: number) => {
```
Comment (hmhealey) [inline packages/mattermost-redux/src/client/client4.ts:873]: Using `UserProfile` above will also let you revert this change

**Review by hmhealey — CHANGES_REQUESTED** [review]

File: packages/mattermost-redux/src/client/client4.ts (hunk 3 above)
Comment (arvinDarmawan) [reply packages/mattermost-redux/src/client/client4.ts]: Hi @hmhealey  I already fix it.

File: components/admin_console/admin_user_card/admin_user_card.tsx (hunk 2 above)
Comment (arvinDarmawan) [reply components/admin_console/admin_user_card/admin_user_card.tsx]: Hi @hmhealey  I already change it.

File: components/admin_console/admin_user_card/admin_user_card.tsx (hunk 1 above)
Comment (arvinDarmawan) [reply components/admin_console/admin_user_card/admin_user_card.tsx]: Hi @hmhealey  I already change it to use type UserProfile.

**Review by hmhealey — APPROVED** [review]

**Review by anurag6713 — APPROVED** [review]
Thanks! LGTM

**Review by furqanmlk — APPROVED** [review]
Test Cases Passed ✅

**Comment by hmhealey** [comment]
Thanks again for the PR, @arvinDarmawan!

**Comment by arvinDarmawan** [comment]
Closes #17245
